### ISSUE TITLE
dataclass false positive `__post_init__`

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#377]
```py
@dataclass
class A:
    def __post_init__(self):
        self.foo = 1  # error
```
